File: src/main/java/org/noxet/noxetserver/util/ControllableTaskSet.java
package org.noxet.noxetserver.util;

import org.bukkit.scheduler.BukkitTask;

import java.util.HashSet;
import java.util.Set;

public class ControllableTaskSet {
    private final Set<BukkitTask> tasks = new HashSet<>();

    public void push(BukkitTask task) {
        tasks.add(task);
    }

    public void abortAll() {
        tasks.forEach(BukkitTask::cancel);
    }
}

File: src/main/java/org/noxet/noxetserver/util/ControllableTaskSet.scala
package org.noxet.noxetserver.util

import org.bukkit.scheduler.BukkitTask

import scala.collection.mutable

class ControllableTaskSet:
  private val tasks: mutable.Set[BukkitTask] = mutable.HashSet()

  def push(task: BukkitTask): Unit = tasks.add(task)

  def abortAll(): Unit = tasks.foreach(_.cancel)
    